express-notes-api/index.js

```diff
@@ -0,0 +1,99 @@
+const fs = require('fs');
+const express = require('express');
+const app = express();
+const data = require('./data');
+const notes = data.notes;
+
+app.use(express.json());
+
+app.get('/api/notes', (req, res) => {
+  const arr = [];
+  for (const property in notes) {
+    arr.push(notes[property]);
+  }
+  res.json(arr);
+  res.sendStatus(200);
+});
+
+app.get('/api/notes/:id', (req, res) => {
+  const id = Number(req.params.id);
+  if (!id || id < 0 || Number.isInteger(id) === false) {
+    res.status(400).json({ error: 'id must be a positive integer' });
+  } else if (!notes[id]) {
+    res.status(404).json({ error: `can not find note with id ${id}` });
+  } else {
+    res.status(200).json(notes[id]);
```

Comment:
Here you don't need to explicitly set the status since you are using status `200`:

```JAVASCRIPT
res.json(notes[id]);
```